flink-table-store-flink/flink-table-store-flink-common/src/main/java/org/apache/flink/table/store/connector/FlinkConnectorOptions.java

```diff
@@ -82,6 +82,20 @@ public class FlinkConnectorOptions {
                                     + "By default, if this option is not defined, the planner will derive the parallelism "
                                     + "for each statement individually by also considering the global configuration.");
 
+    public static final ConfigOption<Boolean> INFER_SCAN_PARALLELISM =
+            ConfigOptions.key("scan.infer-parallelism")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription(
+                            "If it is false, parallelism of source are set by scan.parallelism. "
+                                    + "If it is true, source parallelism is inferred according to splits number (batch mode) or bucket number(streaming mode).");
+
+    public static final ConfigOption<Integer> INFER_SCAN_PARALLELISM_MAX =
+            ConfigOptions.key("scan.infer-parallelism.max")
```

Comment:
Source parallelism is inferred according to splits number (batch mode) or bucket number(streaming mode), therefore when does the user need to config the max of inferred parallelism?

Reply:
In order to prevent too many datafiles from causing excessive parallelism, we also set a max infer parallelism. When the infer parallelism exceeds the setting, use the max parallelism.

Reply:
How does the user configure this config? Meanwhile, the essence of the situation that many datafiles from causing excessive parallelism should be controlled by splits number and bucket number, instead of this config.

Reply:
the  split number is affected by `source.split.target-size` and `source.split.open-file-cost`, for most cases, it is no problem. If the table data is large, the inferred number of splits is too large, but the user does not want this job to occupy too many resources, so set an upper limit, so that the resource utilization of most tasks can be met in a reasonable range, and at the same time, the cluster resources will not be filled up because the split of a job is too large.

refer to `https://github.com/apache/flink/blob/cf358d7d55ca48b9d25e5217006898e3070a85ad/flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveOptions.java#L61`

Reply:
@zhangjun0x01, the implementation of the Hive reference makes sense because hive connector couldn't control the split number, but table store could control splits number and bucket number. I don't think this solution could refer to Hive for the different situation between table table and hive.

Reply:
@zhangjun0x01, +1.